api/working-with-extensions/testing-extension.md

```diff
@@ -9,87 +9,194 @@ MetaDescription: Write tests for your Visual Studio Code extension (plug-in).
 
-## Yo Code test scaffolding
+## Overview
 
-If you are using the [yo code generator](https://github.com/Microsoft/vscode-generator-code), the generated projects include a sample test and instructions for running the tests.
+If you are using the [Yeoman Generator](https://code.visualstudio.com/api/get-started/your-first-extension) to scaffold an extension, integration tests are already setup for you. This page assumes you have read through the [Getting Started](https://code.visualstudio.com/api/get-started/your-first-extension) section and explains how the integration tests work.
 
-**Note**: The documentation below assumes that you created a TypeScript extension but the same also applies for a JavaScript extension. However, some file names may be different.
+You can find the setup for this guide in [helloworld-test-sample](https://github.com/microsoft/vscode-extension-samples/tree/master/helloworld-test-sample).
 
-After you've created a new extension and opened the project in VS Code, you can select the `Extension Tests` configuration from the drop-down at the top of the Debug View.
+## Setup the test command
 
-![launch tests](images/testing-extension/launch-tests.png)
+VS Code provides two arguments to run extension tests. For example:
 
-With this configuration chosen, when you run `Debug: Start` (`kb(workbench.action.debug.start)`), VS Code launches your extension in the `Extension Development Host` instance and runs your tests. Test output goes to the Debug Console where you can see the test results.
+```bash
+code --extensionDevelopmentPath=. --extensionTestsPath=./out/test
+```
 
-![test output](images/testing-extension/test-output.png)
+Although `code` is available on the CLI, it is not available in other environments such as a Continuous Integration machine. Besides, we might want to test against a different version of VS Code. Let us first setup a script to download VS Code and run the extension test using `vscode-test`:
+
+- `npm install -D vscode-test`
+- Write a test script:
+
+```ts
+import * as path from 'path';
+
+import { runTests } from 'vscode-test';
+
+async function go() {
+  try {
+    const extensionPath = path.resolve(__dirname, '../../');
+    const testRunnerPath = path.resolve(__dirname, './testRunner.js');
+
+    // Download VS Code, unzip it and run the integration test
+    await runTests({
+      // The folder containing the Extension Manifest package.json
+      // Passed to `--extensionDevelopmentPath`
+      extensionPath,
+      // The path to test runner
+      // Passed to --extensionTestsPath
+      testRunnerPath
+    });
```

Comment:
I'd move the comments up to the `const` declarations, so as to keep this `await` simpler. Something like this:

```ts
    // The folder containing the Extension Manifest package.json
    // Passed to `--extensionDevelopmentPath`
    const extensionPath = path.resolve(__dirname, '../../');

    // The path to test runner
    // Passed to --extensionTestsPath
    const testRunnerPath = path.resolve(__dirname, './testRunner.js');

     // Download VS Code, unzip it and run the integration test
    await runTests({ extensionPath, testRunnerPath });
```